pkg/importer/spanner/sysl.arrai

```diff
@@ -0,0 +1,138 @@
+        (mode = 2 && entity.foreign_keys):
+            'foreign_keys=[' ++ //seq.join(', ',(entity.foreign_keys => \keys '["constraint:' ++ keys.constraint_name ++
+            '","columns:' ++ //seq.join(', ', (keys.foreign_keys => .attribute) orderby .) ++ '"]' ) orderby .) ++ ']',
+        let relatedIndices = ((model.indexes orderby .) where .@item.table_name = entity.name); (mode = 3 && relatedIndices):
+            'indexes=[' ++ //seq.join(', ', ((model.indexes orderby .) where .@item.table_name = entity.name) >>
+            '"name:' ++ .table_name ++
+            '","unique:' ++ (cond {(.unique = 'UNIQUE'):'true', _:'false'}) ++
+            '","null_filtered:' ++ (cond {(.nullfiltered = 'NULL_FILTERED'):'true', _:'false'}) ++
+            '","key_parts:' ++ //seq.join(', ', .key_part orderby .) ++
+            cond {(.storing_col): '","storing:' ++ (//seq.join(', ',.storing_col))}  ++
+            cond {(.interleaved_table): '","interleave_in:' ++ .interleaved_table}) ++ '"]',
+        _: '',
+    };
+
+# allEntityPatterns generates the pattern string for an entity
+let allEntityPatterns = \entity \model
+    $`${entityPatternToken(entity, model, '[')}${cond {
+        entityPatternToken(entity, model, true):
+        //seq.join(', ',[entityPattern(entity, model, 0), entityPattern(entity, model, 1),
+        entityPattern(entity, model, 2), entityPattern(entity, model, 3)] where .@item != '')
+    }}${entityPatternToken(entity, model, ']')}`;
+
+# allAttributePatterns generates the pattern string for an entity attribute
+let allAttributePatterns = \entity \attr
+    $`${attributePatternToken(entity,attr,'[')}${cond {
+        attributePatternToken(entity,attr,true):
+        //seq.join(', ',[attributePattern(entity,attr,2), attributePattern(entity,attr,0), attributePattern(entity,attr,1),
+        attributePattern(entity,attr,3), attributePattern(entity,attr,4)] where .@item != '')
+    }}${attributePatternToken(entity,attr,']')}`;
+
+# typeInfo generates the type info for an attribute
+let typeInfo = \entity \attr \type \isArray
+    let refAttr = entity.foreign_keys => ((.foreign_keys where .attribute = attr.name) =>
+        .reference_attribute orderby .);
+    let refTable = entity.foreign_keys => ((.foreign_keys where .attribute = attr.name) =>
+        .reference_table orderby .);
+    let fkAttr = (refAttr where . != {});
+    let fkTable = (refTable where . != {});
+    cond {
+        fkAttr | fkTable:  //seq.join('', (fkTable => //seq.join('', .) orderby .)) ++ '.'
+            ++ //seq.join('', (fkAttr => //seq.join('', .) orderby .)),
+        isArray: 'sequence of ' ++ type,
+        _: type,
+    };
+
+# transformModel translates the empty model into sysl file
+let transformModel = \model \package
+    # sysl specification
+    # https://github.com/anz-bank/sysl/blob/master/pkg/sysl/sysl.proto
+    $`
+        ########   THIS IS AUTOGENERATED BY sysl   ########
```

Comment:
Nit: The general Golang convention is:
```suggestion
        // Code generated by sysl. DO NOT EDIT.
```

Reply:
It is sysl being generated here.

Reply:
My point is not about the language, but the style of the comment. I realise now that files generated by `sysl import` have their own style of header comment.

However `pkg/importer/writer.go` adds this header to the beginning of the generated source, so if this arr.ai script is being used within the context of `sysl import`, then this script shouldn't be adding a header comment like this at all.

Reply:
Other importers use `pkg/importer/writer.Write()` to write out their Sysl string, and `Write` prepends this autogenerated header. Ideally use that for consistency (export `writeHeader` if necessary).